Sources/SwiftGD/Image.swift

```diff
@@ -118,21 +118,28 @@ public class Image {
     public func renderText(
         _ text: String, from: Point, fontList: [String], color: Color, size: Double, angle: Angle = .zero
     ) -> (upperLeft: Point, upperRight: Point, lowerRight: Point, lowerLeft: Point) {
+        /// Notes on `gdImageStringFT`:
+        /// - it returns an Tuple of empty `Point`s if there is nothing to render or no valid fonts
+        /// - `gdImageStringFT` accepts a semicolon delimited list of fonts.
+        /// - `gdImageStringFT` expects pointers to `text` and `fontList` values
+        guard !text.isEmpty,
+              !fontList.isEmpty,
+              var textCChar = text.cString(using: .utf8),
+              var joinedFonts = fontList.joined(separator: ";").cString(using: .utf8) else {
+                  return (upperLeft: .zero, upperRight: .zero, lowerRight: .zero, lowerLeft: .zero)
+        }
         let red = Int32(color.redComponent * 255.0)
         let green = Int32(color.greenComponent * 255.0)
         let blue = Int32(color.blueComponent * 255.0)
         let alpha = 127 - Int32(color.alphaComponent * 127.0)
         let internalColor = gdImageColorAllocateAlpha(internalImage, red, green, blue, alpha)
         defer { gdImageColorDeallocate(internalImage, internalColor) }
 
-        // `gdImageStringFT` accepts a semicolon delimited list of fonts.
-        let fontList = fontList.joined(separator: ";")
-
         // `gdImageStringFT` returns the text bounding box, specified as four
         // points in the following order:
-        // lower left, lower right, upper right, and upper left corner.
+        // upper left, upper right, lower right, and lower left corner.
```

Comment:
This update is not correct:
- `renderText` returns `upper left, upper right, lower right, and lower left corner.`
-  `gdImageStringFT` returns `lower left, lower right, upper right, and upper left corner.`

The comment is about `gdImageStringFT`.

Reply:
@zntfdr @twostraws My apologies. I have a new patch ( #33 ) in to address the mistake.

Reply:
@mcritz Thank you for the quick patch ❤️